src/data/upload.ts

```diff
@@ -122,10 +123,15 @@ export function end(): void {
     active = false;
 }
 
-function upload(last: boolean = false): void {
+function upload(final: boolean = false): void {
     memory.compute();
     target.compute();
     metric.compute();
+
+    // Treat this as the last payload only if final boolean was explictly set to true
+    // In some edge cases, certain third party scripts (e.g. script error monitoring) could inject bogus arguments in the end to capture
+    // valid stack traces. In those cases, we don't want a random injected object (e.g. { "key": "value" }) to be mistaken for true.
```

Comment:
>those [](start = 30, length = 5)

Can you expand a bit to describe what "inject bogus arguments" means and how it relates to capturing valid stack traces. Do you have an example of such a third party script?

Reply:
A sample example is: https://www.npmjs.com/package/raven-js